src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -26,107 +34,167 @@
-                    "'format' query parameter can have one of the next values: %s", trackingPixels.keySet()));
+    private Future<Boolean> isAccountEventEnabled(String accountId) {
+        return applicationSettings.getAccountById(accountId, timeoutFactory.create(DEFAULT_TIMEOUT))
+                .map(Account::getEventsEnabled)
+                .otherwise(throwable -> fallbackResult(accountId, throwable));
+    }
+
+    private void handleEvent(AsyncResult<Boolean> isEnabledResult, boolean isAnalyticsEnabled, String format,
+                             RoutingContext context) {
+        final HttpServerResponse response = context.response();
+        if (isEnabledResult.result()) {
+            if (isAnalyticsEnabled) {
+                analyticsReporter.processEvent(makeNotificationEvent(context));
+                respondWithOkStatus(response, format);
+            }
+        } else {
+            respondWithUnauthorized(response, "Given 'accountId' is not supporting the event");
         }
     }
 
-    private void respondWithBadStatus(RoutingContext context, String message) {
-        context.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
-                .end(String.format("Request is invalid: %s", message));
+    private static NotificationEvent makeNotificationEvent(RoutingContext context) {
+        final MultiMap queryParameters = context.request().params();
+        final NotificationEvent.Type type = toNotificationType(queryParameters.get(TYPE_PARAMETER));
+        final String bidId = queryParameters.get(BID_ID_PARAMETER);
+        final String accountId = queryParameters.get(ACCOUNT_PARAMETER);
+        final HttpContext httpContext = HttpContext.from(context);
+        return NotificationEvent.of(type, bidId, accountId, httpContext);
     }
 
-    private void respondWithOkStatus(RoutingContext context, String format) {
-        if (format != null) {
-            final TrackingPixel trackingPixel = trackingPixels.get(format);
-            context.response()
-                    .putHeader(HttpHeaders.CONTENT_TYPE, trackingPixel.getContentType())
+    private static NotificationEvent.Type toNotificationType(String type) {
+        return Objects.equals(WIN_TYPE, type) ? NotificationEvent.Type.win : NotificationEvent.Type.imp;
+    }
+
+    private void respondWithOkStatus(HttpServerResponse response, String format) {
+        if (StringUtils.equals(format, IMP_TYPE)) {
+            response.putHeader(HttpHeaders.CONTENT_TYPE, trackingPixel.getContentType())
                     .end(Buffer.buffer(trackingPixel.getContent()));
         } else {
-            context.response().end();
+            response.end();
+        }
+    }
+
+    private Boolean fallbackResult(String accountId, Throwable throwable) {
+        if (!(throwable instanceof PreBidException)) {
+            logger.warn(String.format("Error when retrieving account with %s id", accountId), throwable);
```

Comment:
We can use built-in placeholders for logger, like `{0}`. So, no reason for `String.format`.